packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -997,11 +999,27 @@ class ProjectFileInvalidator {
         printError('Warning: updated dependencies detected. The Flutter application will require a restart to safely use new packages.');
       }
       _packagesUpdateTime = newPackagesUpdateTime;
+      final bool hadBuilders =_hasBuilders;
+      _hasBuilders = _flutterProject?.hasBuilders ?? false;
+      if (hadBuilders != _hasBuilders) {
+        printError('Warning: Added builders detected. The Flutter runner will require a restart to use them.');
+      }
     }
     final List<String> invalidatedFiles = <String>[];
+    // To support codegen for arbitrary packages, this logic will need to be
+    // expanded for each package.
     for (String packageName in _packageMap.keys) {
       final Uri packageUri =_packageMap[packageName];
-      _scanDirectory(packageUri, invalidatedFiles);
+      if (_flutterProject != null && _flutterProject.manifest.appName == packageName && _hasBuilders) {
+        final Uri generatedUri = _flutterProject
+          .generated
+          .childDirectory('lib')
+          .uri;
+        _scanDirectory(packageUri, invalidatedFiles);
+        _scanDirectory(generatedUri, invalidatedFiles);
+      } else {
+        _scanDirectory(packageUri, invalidatedFiles);
```

Comment:
Looks like you could avoid the `else` clause by moving `_scanDirectory(packageUri, invalidatedFiles);` above the `if` block. It has to run either way.

Reply:
This is no longer necessary